examples/simulator-lang-rps-2-rps/client.mjs

```diff
@@ -0,0 +1,38 @@
+import * as lang from '@reach-sh/simulator-lang';
+import * as assert from 'assert';
+
+const main = async () => {
+  console.log("Init Testing!")
+  const s = new lang.Scenario();
+  await s.init();
+  const pi = await s.pingServer();
+  const alice = s.participants[0];
+  const bob = s.participants[1];
+  const consensus = s.consensus;
+  // init Alice
+  await alice.init();
+  // init Bob
+  await bob.init();
+  // Alice interactively gets her hand (0)
+  await (await alice.getNextAction()).resolve(0);
+  // Alice's hand (0) is published
+  await (await consensus.getNextAction()).resolve(0);
+  // Alice observes that her hand is published
+  await (await alice.getNextAction()).resolve(-4);
+  // Bob observes that Alice's hand is published
+  await (await bob.getNextAction()).resolve(-4);
+  // Bob interactively gets his hand (1)
+  await (await bob.getNextAction()).resolve(1);
+  // Bob's hand (1) is published
+  await (await consensus.getNextAction()).resolve(1);
+  // Alice observes that Bob's hand is published
+  await (await alice.getNextAction()).resolve(-4);
+  // Bob observes that his hand is published
+  await (await bob.getNextAction()).resolve(-4);
+  const r = await alice.getStatus();
```

Comment:
I don't like how stateful this is... I think I want it to be functional where the nodes of the graph are represented in the program and if you want to write things imperatively then you either do that yourself by doing
```
s = s.who(Mary).act(2);
s = s.who(Jane).act(3);
```
or maybe there's an option to have
```
const s = makeImperativeScenario(fromState);
s.who(Mary).act(2);
s.who(Jane).act(3);
```

Reply:
My overall point is that this program doesn't look compelling as being a simple way to understand your code and it isn't testing anything other than that the thing doesn't crash. I want to be able to test that Bob wins or that when there's a timeout, Alice gets her money back. We want to show the value of the simulator.

Reply:
for sure :)